Rx433.cpp

```diff
@@ -62,63 +62,58 @@ bool Handle(std::vector<Pulse> buf) {
 
 void ICACHE_RAM_ATTR rxISR() {
   static int last_changed = 0, sync_pulse_us = 0;
+  static int last_state = 0;
   static std::vector<rx433::Pulse> pulse_stream;
-  bool pulse_stream_done = false;
-  if (pulse_stream_queue.size() > 10) return;
   int now = micros();
+  int state = digitalRead(rxPin_);
+  int delta_us = now - last_changed;
+
+  // Ignore very short glitches that we missed.
+  if (state == last_state) return;
+  last_state = state;
+
+  // Streamed glich filter
+  // Ignore any pulse (1 or 0) shorter than kGlitchUs.
+  if (!pulse_stream.empty() && delta_us < kGlitchUs) {
+    last_changed -= pulse_stream.back().delta_us;
+    pulse_stream.pop_back();
+    return;
+  }
+  last_changed = now;
+
+  // Queue is full, drop pulse
+  if (pulse_stream_queue.size() > 10) return;
+
+  // Create pulse
   // Need to invert as p represents state before transition.
-  Pulse p = {now, now-last_changed, digitalRead(rxPin_)};
+  Pulse p = {now, delta_us, state};
+
+  bool pulse_stream_done = false;
   if (pulse_stream.empty() && IsSync(p)) {
     sync_pulse_us = p.delta_us;
     pulse_stream.push_back(p);
   } else if (p.delta_us * 4 > 3 * sync_pulse_us) {
     pulse_stream_done = true;
-  } else {
-    if (pulse_stream.size() > kMaxPulseLen) {
-      pulse_stream_done = true;
-    } else if (!pulse_stream.empty()) {
-      pulse_stream.push_back(p);
-    }
+  } else if (pulse_stream.size() > kMaxPulseLen) {
+    pulse_stream_done = true;
+  } else if (!pulse_stream.empty()) {
+    // Put pulses after sync in stream.
+    pulse_stream.push_back(p);
   }
-  if (pulse_stream_done && pulse_stream.size() > kMinPulseLen) {
-    pulse_stream_queue.push_back(std::move(pulse_stream));
+
+  if (pulse_stream_done) {
+    if (pulse_stream.size() >= kMinPulseLen) {
+      // A complete pulse stream to handle.
+      pulse_stream_queue.push_back(std::move(pulse_stream));
+    }
     pulse_stream_done = false;
-    // Need to reset as it was move:d above.
+    // Need to reset also when it was move:d above.
     pulse_stream.clear();
     if (IsSync(p)) {
       sync_pulse_us = p.delta_us;
-      pulse_stream.push_back(p);
+      pulse_stream.push_back(std::move(p));
```

Comment:
Nit: you don't gain anything with this move as p is stack allocated

Reply:
Done